src/MongoDB.Driver.Core/Core/Operations/ChangeStreamCursor.cs

```diff
@@ -155,7 +154,17 @@ private IEnumerable<TDocument> DeserializeDocuments(IEnumerable<RawBsonDocument>
 
             if (lastRawDocument != null)
             {
-                _changeStreamOperation.ResumeAfter = lastRawDocument["_id"].DeepClone().AsBsonDocument;
+                var resumeToken = lastRawDocument["_id"].DeepClone().AsBsonDocument;
+
+                if (_changeStreamOperation.StartAfter != null && _changeStreamOperation.ResumeAfter == null)
```

Comment:
I've changed this logic. In the previous implementation, I just set `StartAfter` to `null`.
But it looks like here we can reinitialize field which was originally initialized when we created `watch`